lk2nd/project/base.mk

```diff
@@ -54,4 +54,11 @@ DEFINES := $(filter-out USER_FORCE_RESET_SUPPORT=1, $(DEFINES))
 # Allow flashing independent of battery voltage
 DEFINES := $(filter-out CHECK_BAT_VOLTAGE=1, $(DEFINES))
 
+# lk2nd provides its own mainline-friendly partial-goods implementation
+ifeq ($(ENABLE_PARTIAL_GOODS_SUPPORT), 1)
```

Comment:
Can you make this conditional to `msm8909` for now? We don't handle msm8996 in this implementation yet and I'm unsure on which devices partial-goods would be relevant for msm8996.

Reply:
This condition disables the downstream-specific partial goods code, which is needed on all platforms in order to avoid problems with that code. The lk2nd partial goods replacement already has an `#if` condition for msm8909 and does nothing on other platforms, so I'm not sure if adding another condition here would improve anything.

Reply:
I specifically want it to fail on MSM8996 because if someone reports that we know that these devices exist (rather than failing to do the necessary patching there). I suspect most MSM8996 devices don't have any "partial goods" restriction and in this case the original code does actually nothing and doesn't fail like on MSM8905.